Azure-Samples/azure-search-openai-demo
# Render entire PDFs instead of single pages

## Purpose

Fixes #819 

According to various user studies and requests, users would generally prefer to see the entire PDF, so that they can see the context, but of course, they'd also like the PDF to start off on the relevant page. Fortunately, browsers support a pseudo-standard of "#page=N" for jumping to a particular page.

This PR changes prepdocs.py to skip the step of uploading individual pages and link to #page=N instead of filename-N.pdf, and then makes sure the content_file route can still load those.

<img width="1713" alt="Screenshot 2023-10-20 at 4 07 41 PM" src="https://github.com/Azure-Samples/azure-search-openai-demo/assets/297042/fdf6e550-3165-453d-8fab-7bb631b5f97a">


I added tests for the content_file route as well. They're running slowly now (though passing), am hoping to get guidance from Azure SDK team as to improving my mocking approach.


## Does this introduce a breaking change?

This should still work with indices that have individual pages.


```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test with a fresh environment, azd up, ask a question, click on citation, see it jump
* Test with an old environment, dont re-index, ask a question, click on citation, see it load whole page
